providers/flagd/pkg/service/in_process/service.go

```diff
@@ -0,0 +1,288 @@
+			ProviderResolutionDetail: of.ProviderResolutionDetail{
+				ResolutionError: mapError(err),
+				Reason:          of.Reason(reason),
+				Variant:         variant,
+				FlagMetadata:    metadata,
+			},
+		}
+	}
+
+	return of.StringResolutionDetail{
+		Value: value,
+		ProviderResolutionDetail: of.ProviderResolutionDetail{
+			Reason:       of.Reason(reason),
+			Variant:      variant,
+			FlagMetadata: metadata,
+		},
+	}
+}
+
+func (i *InProcess) ResolveFloat(ctx context.Context, key string, defaultValue float64,
+	evalCtx map[string]interface{}) of.FloatResolutionDetail {
+	value, variant, reason, metadata, err := i.evaluator.ResolveFloatValue(ctx, "", key, evalCtx)
+	if err != nil {
+		return of.FloatResolutionDetail{
+			Value: defaultValue,
+			ProviderResolutionDetail: of.ProviderResolutionDetail{
+				ResolutionError: mapError(err),
+				Reason:          of.Reason(reason),
+				Variant:         variant,
+				FlagMetadata:    metadata,
+			},
+		}
+	}
+
+	return of.FloatResolutionDetail{
+		Value: value,
+		ProviderResolutionDetail: of.ProviderResolutionDetail{
+			Reason:       of.Reason(reason),
+			Variant:      variant,
+			FlagMetadata: metadata,
+		},
+	}
+}
+
+func (i *InProcess) ResolveInt(ctx context.Context, key string, defaultValue int64,
+	evalCtx map[string]interface{}) of.IntResolutionDetail {
+	value, variant, reason, metadata, err := i.evaluator.ResolveIntValue(ctx, "", key, evalCtx)
+	if err != nil {
+		return of.IntResolutionDetail{
+			Value: defaultValue,
+			ProviderResolutionDetail: of.ProviderResolutionDetail{
+				ResolutionError: mapError(err),
+				Reason:          of.Reason(reason),
+				Variant:         variant,
+				FlagMetadata:    metadata,
+			},
+		}
+	}
+
+	return of.IntResolutionDetail{
+		Value: value,
+		ProviderResolutionDetail: of.ProviderResolutionDetail{
+			Reason:       of.Reason(reason),
+			Variant:      variant,
+			FlagMetadata: metadata,
+		},
+	}
+}
+
+func (i *InProcess) ResolveObject(ctx context.Context, key string, defaultValue interface{},
+	evalCtx map[string]interface{}) of.InterfaceResolutionDetail {
+	value, variant, reason, metadata, err := i.evaluator.ResolveObjectValue(ctx, "", key, evalCtx)
+	if err != nil {
+		return of.InterfaceResolutionDetail{
+			Value: defaultValue,
+			ProviderResolutionDetail: of.ProviderResolutionDetail{
+				ResolutionError: mapError(err),
+				Reason:          of.Reason(reason),
+				Variant:         variant,
+				FlagMetadata:    metadata,
+			},
+		}
+	}
+
+	return of.InterfaceResolutionDetail{
+		Value: value,
+		ProviderResolutionDetail: of.ProviderResolutionDetail{
+			Reason:       of.Reason(reason),
+			Variant:      variant,
+			FlagMetadata: metadata,
+		},
+	}
+}
+
+func (i *InProcess) EventChannel() <-chan of.Event {
+	return i.events
+}
+
+// mapError is a helper to map evaluation errors to OF errors
+func mapError(err error) of.ResolutionError {
+	switch err.Error() {
+	case model.FlagNotFoundErrorCode, model.FlagDisabledErrorCode:
```

Comment:
Flag disabled shouldn't be an error, I think. It's a reason, and the deftault should be returned without an error being indicated.

See reasons in the resolution details: https://openfeature.dev/specification/types#resolution-details, and the [FAQ](https://openfeature.dev/docs/reference/concepts/provider#why-isnt-there-an-error-code-for-disabled-flags) on this situation.

Reply:
hmm, I think we need to revisit this issue. We have different handling of disabled state.

- Flagd core : generates an error [^1]
- Java provider : consider this as an error [^2]
- JS provider : consider this as NOT an error [^3]

And the go provider currently consider this as an error - same for current in-process go provider 

[^1]: https://github.com/open-feature/flagd/blob/main/core/pkg/service/flag-evaluation/flag_evaluator.go#L318-L319
[^2]: https://github.com/open-feature/java-sdk-contrib/blob/main/providers/flagd/src/main/java/dev/openfeature/contrib/providers/flagd/resolver/process/InProcessResolver.java#L159-L162
[^3]: https://github.com/open-feature/js-sdk-contrib/blob/main/libs/shared/flagd-core/src/lib/flagd-core.ts#L85-L91

Reply:
Ok, ya, we should be consistent here. The reasoning is that disabled flags aren't really an exceptional circumstance (it's a supported configuration in most systems) therefore it shouldn't involve an exception-based control flow.

But anyway, I agree it seems to be a bigger issue than just this PR, so do what you can here and I won't consider it a blocker. We can open other issues where it's relevant.